qiita_pet/test/rest/test_study_samples.py

```diff
@@ -91,21 +209,75 @@ def test_patch_sample_ids_exist_incomplete_metadata(self):
         self.assertEqual(obs, exp)
 
     def test_patch_sample_ids_complete_metadata_and_unknown_metadata(self):
-        body = _sample_creator(['1.SKM8.640201', 'blank.a1'])
+        current = self._get_sample_categories(1)
+        # If no new categories, both new and existing samples should succeed.
+        # 640201 is an existing sample. blank.a1 is a new sample
+        body = _sample_creator(['1.SKM8.640201',
+                                'blank.a1'], categories=current)
+        response = self.patch('/api/v1/study/1/samples', headers=self.headers,
+                              data=body, asjson=True)
+        self.assertEqual(response.code, 201)
+        # successful response should be empty string
+        self.assertEqual(response.body, b'')
+
+        # If new categories are added, patch() should succeed.
+        # New and existing samples should have new categories.
+        # 640201 is an existing sample. blank.a2 is a new sample
+        body = _sample_creator(['1.SKM8.640201',
+                                'blank.a2'], categories=current)
+        # body['blank.a2']['DOES_NOT_EXIST'] will be '', not None.
```

Comment:
is this True? I think is foo, no? Same for WHAT/bar ...

Reply:
Nope. Confirmed Friday that whenever you add a new category like we're doing here, all samples mentioned in the body will have their values set to the value you explicitly define or ''. All samples NOT mentioned in the body will have their values set to None.